tensorboard/webapp/runs/views/runs_table/sorting_utils_test.ts

```diff
@@ -0,0 +1,284 @@
+              id: 'row 2 id',
+              name: '2/myrun',
+            },
+            {
+              id: 'row 3 id',
+              name: '10/myrun',
+            },
+          ],
+          {
+            order: SortingOrder.ASCENDING,
+            name: 'name',
+          }
+        )
+      ).toEqual([
+        {
+          id: 'row 1 id',
+          name: '1/myrun',
+        },
+        {
+          id: 'row 2 id',
+          name: '2/myrun',
+        },
+        {
+          id: 'row 3 id',
+          name: '10/myrun',
+        },
+      ]);
+    });
+
+    it('sorts runs with purely numeric run names before runs with leading numbers', () => {
+      expect(
+        sortTableDataItems(
+          [
+            {
+              id: 'row 1 id',
+              name: '0',
+            },
+            {
+              id: 'row 2 id',
+              name: '0/myrun2',
+            },
+            {
+              id: 'row 3 id',
+              name: '0/myrun1',
+            },
+          ],
+          {
+            order: SortingOrder.ASCENDING,
+            name: 'name',
+          }
+        )
+      ).toEqual([
+        {
+          id: 'row 1 id',
+          name: '0',
+        },
+        {
+          id: 'row 3 id',
+          name: '0/myrun1',
+        },
+        {
+          id: 'row 2 id',
+          name: '0/myrun2',
+        },
+      ]);
+    });
+
+    it('sorts runs with string names', () => {
+      expect(
+        sortTableDataItems(
+          [
+            {
+              id: 'row 1 id',
+              name: 'aaa',
+            },
+            {
+              id: 'row 2 id',
+              name: 'bbb',
+            },
+            {
+              id: 'row 3 id',
+              name: 'ccc',
+            },
+          ],
+          {
+            order: SortingOrder.ASCENDING,
+            name: 'name',
+          }
+        )
+      ).toEqual([
+        {
+          id: 'row 1 id',
+          name: 'aaa',
+        },
+        {
+          id: 'row 2 id',
+          name: 'bbb',
+        },
+        {
+          id: 'row 3 id',
+          name: 'ccc',
+        },
+      ]);
+    });
+
+    it('shows runs without numbers before runs with numbers', () => {
+      expect(
+        sortTableDataItems(
+          [
+            {
+              id: 'row 1 id',
+              name: 'aaa',
+            },
+            {
+              id: 'row 2 id',
+              name: '1aaa',
+            },
+            {
+              id: 'row 3 id',
+              name: '2bbb',
+            },
+          ],
+          {
+            order: SortingOrder.ASCENDING,
+            name: 'name',
+          }
+        )
+      ).toEqual([
+        {
+          id: 'row 1 id',
+          name: 'aaa',
+        },
+        {
+          id: 'row 2 id',
+          name: '1aaa',
+        },
+        {
+          id: 'row 3 id',
+          name: '2bbb',
+        },
+      ]);
+    });
+
+    it('places undefined values at the end', () => {
```

Comment:
If you are going to have the ability to place undefined values before(which I am not sure we really need). Then you should have a test for it.

Reply:
We kinda have that already, that is what `sorts runs with purely numeric run names before runs with leading numbers` is checking

Reply:
I can see how that test along with `returns undefined when a non numeric value is provided` kind of tests for it.